sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -543,13 +543,14 @@ object LikeSimplification extends Rule[LogicalPlan] {
   private val equalTo = "([^_%]*)".r
 
   def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
-    case Like(input, Literal(pattern, StringType), escapeChar) =>
+    case l @ Like(input, Literal(pattern, StringType), escapeChar) =>
       if (pattern == null) {
         // If pattern is null, return null value directly, since "col like null" == null.
         Literal(null, BooleanType)
       } else {
         val escapeStr = String.valueOf(escapeChar)
         pattern.toString match {
+          case _ if escapeChar == '_' || escapeChar == '%' => l
```

Comment:
Please add tests in `LikeSimplificationSuite`, too? Btw, does this simplification work correctly if `escapeChar` is not a default one other than `_` and `%`?

Reply:
You are right! I think this rule works correctly only if `pattern` is valid and `escapeChar` does not escape itself. I will fix this and change the title/description.

Reply:
oh... I see...